src/delibird/work/mock.py

```diff
@@ -118,12 +120,16 @@ def write_directory(directory, columns, row_number, batch_size=1024 * 1024):
         row_number (int): row number
         batch_size (int): write batch size
     """
-    # multiprocess starmap
+    # adjust a reasonable batch_size which would not cause an OutOfMemoryError
+    sample_dict_list = gen_dict(columns, 1)
+    safe_batch_size = simple_batch_size(sample_dict_list, min(row_number, batch_size))
+    
     with Pool(processes=cpu_count()) as pool:
+        # multiprocess starmap
         # init batch number list
-        record_list = [batch_size for _ in range(row_number // batch_size)]
-        if row_number % batch_size:
-            record_list.append(row_number % batch_size)
+        record_list = [safe_batch_size for _ in range(row_number // safe_batch_size)]
```

Comment:
yes, this is the key
check other places if had the same problems
Also, the "simple" name is not so good to express the actual meaning
use "opti"(optimization) or "cal"(calculate)  may be more suitable